### PR TITLE
Add Ansible Runner Task

### LINKED ISSUE #463: Add a Task to interact with Kubernetes Cluster with Ansible
The [Ansible Kubernetes](https://github.com/ansible-collections/community.kubernetes) provides a no `kubectl` way to interact with Kubernetes clusters using Ansible.  

A Task that can help to run Ansible playbook  typically targeted in performing Kubernetes cluster interaction will be of great value.

### PR DESCRIPTION
**Pull request by kameshsampath**
Signed-off-by: Kamesh Sampath <ksampath@redhat.com>

# Changes

A new Task to run Ansible Playbook, that can help performing Kubernetes cluster interaction.  Resolves #463 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [* ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ *] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [*] Yaml file complies with [yamllint](https://github.com/adrienverge/yamllint) rules.
- [ ] Complies with [Catalog Orgainization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [ ] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [* ] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [*] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [*] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [*] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Comment by vinamra28** [comment]
Thanks @kameshsampath for adding this. Can we change the PR title to `Add Ansible Kubernetes Task` instead of `(fix) Ansible Kubernetes Task` as this PR is adding a new task to the catalog instead of providing a fix to the existing one

File: task/ansible-kubernetes/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM registry.access.redhat.com/ubi8/python-36 
+
+ADD requirements.txt .
+ADD requirements.yml .
+
+LABEL ANSIBLE_VERSION 2.9.10 \
+     ANSIBLE_KUBERNETES 0.11.0
+
+USER root
```
[the remark this answers is not in the document]
Comment (kameshsampath) [reply task/ansible-kubernetes/Dockerfile:9]: If I need to setting the alternatives to default to python3 I need to, and I switch back to User 1001 after that. Is that still an issue ?

File: task/ansible-kubernetes/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+FROM registry.access.redhat.com/ubi8/python-36
```
[the remark this answers is not in the document]
Comment (kameshsampath) [reply task/ansible-kubernetes/Dockerfile:1]: IMHO no, hence I have to build one

**Comment by kameshsampath** [comment]
> can we please take reference from the task [blue-green-deploy](https://github.com/tektoncd/catalog/tree/master/task/blue-green-deploy/0.1) or any other task from catalog to improve the task structure?

What you mean here @vinamra28 , moving things under 0.1 folder ?

File: task/ansible-kubernetes/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (siamaksade) [reply task/ansible-kubernetes/Dockerfile:1]: Isn't `docker.io/ansible/ansible-runner` what you need?

[the remark this answers is not in the document]
Comment (kameshsampath) [reply task/ansible-kubernetes/Dockerfile:1]: Ah! @siamaksade  thanks, last time when I checked runner it was using Py2 so thought not to use, glad its there now will update it.

[the remark this answers is not in the document]
Comment (kameshsampath) [reply task/ansible-kubernetes/Dockerfile:1]: I can change the base image, but then I need to install the collections `community.kubernetes` there, wondering thats to do :D

[the remark this answers is not in the document]
Comment (kameshsampath) [reply task/ansible-kubernetes/Dockerfile]: @siamaksade @chmouel  - updated it to use the ansible runner

File: task/ansible-kubernetes/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (kameshsampath) [reply task/ansible-kubernetes/Dockerfile]: latest commit resolves this

**Comment by kameshsampath** [comment]
> can we please take reference from the task [blue-green-deploy](https://github.com/tektoncd/catalog/tree/master/task/blue-green-deploy/0.1) or any other task from catalog to improve the task structure?

the task has been updated to reflect the new structure and also renamed the task to be called `ansible-runner`

File: task/ansible-runner/0.1/ansible-runner.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      default:
+        - --help
+  steps:
+    - name: requirements
+      image: docker.io/ansible/ansible-runner
+      script: |
+        #!/bin/bash 
+        set +x 
+        set -e
+        echo "Jai Guru"
```
[the remark this answers is not in the document]
Comment (kameshsampath) [reply task/ansible-runner/0.1/ansible-runner.yaml:40]: Oh yeah, will remove it

File: task/ansible-runner/0.1/ansible-runner.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        fi
+      workingDir: '$(workspaces.runner-dir.path)/$(params.project-dir)'
+
+    - name: run-playbook
+      image: docker.io/ansible/ansible-runner
+      script: '$(params.script)'
+      args:
+        - $(params.args)
+        - $(params.project-dir)
+      workingDir: '$(workspaces.runner-dir.path)'
```
[the remark this answers is not in the document]
Comment (kameshsampath) [reply task/ansible-runner/0.1/ansible-runner.yaml:63]: I would keep it separate for clarity ?

**Comment by kameshsampath** [comment]
@vinamra28  - squashed commits to one and pushed back with updated review comments and lint checks

**Comment by chmouel** [comment]
Is there any chance we can have a test for this ?

We could have a hello world playbook getting generated in a step (ie: https://codingbee.net/ansible/ansible-a-hello-world-playbook) and have  the task test if it works.

We are trying to ensure we have as much test coverage for our tasks as possible

**Comment by kameshsampath** [comment]
> Is there any chance we can have a test for this ?
> 

Is it OK if we add test to verify whats being done by `ansible-runner` task , I mean saying  deploy an app to k8s with ansible-runner and then query to verify using another kubectl/oc client task ?

**Comment by chmouel** [comment]
> Is it OK if we add test to verify whats being done by ansible-runner task , I mean saying deploy an app to k8s with ansible-runner and then query to verify using another kubectl/oc client task ?

yeah exactly but we don't have to  the last part to query to verify if the task was done properly, we want focus on the testing the actual task not necessary if the playbook has run properly.. (and this add a certain level of complexity)

**Comment by kameshsampath** [comment]
> > Is it OK if we add test to verify whats being done by ansible-runner task , I mean saying deploy an app to k8s with ansible-runner and then query to verify using another kubectl/oc client task ?
> 
> yeah exactly but we don't have to the last part to query to verify if the task was done properly, we want focus on the testing the actual task not necessary if the playbook has run properly.. (and this add a certain level of complexity)


@chmouel - added the tests to verify the taskrun.

**Comment by chmouel** [comment]
Good job, thanks 👍

/lgtm
/approve

**Pull request merged**